hadoop-hdds/docs/content/feature/FairCallQueue.md

```diff
@@ -0,0 +1,100 @@
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+FairCallQueue
+===
+
+This document contains information for setting up the `FairCallQueue` feature with Ozone 
+and testing it in a docker based dev cluster. In order for `FairCallQueue` to be enabled and used, 
+Hadoop RPC must be used as transport protocol for OM - S3G communication. There is no implementation for gRPC.
+
+There is a custom `IdentityProvider` implementation for Ozone that must be specified in the configuration, otherwise
+there is no S3G impersonation which makes the `FairCallQueue` ineffective since it's only reading one user.
+
+## Configuration
+
+There must be a port specified to which the OM will forward any activity 
+and the `FailCallQueue` and `DecayRpcScheduler` will listen to. 
+Furthermore, this port will have to be part of every configuration name.
+
+Port used for below examples : 9862
+
+```XML
+<property>
+   <name>ozone.om.address</name>
+   <value>OMDomain:9862</value>
+</property>
+
+<property>
+    <name>ozone.om.s3.grpc.server_enabled</name>
+    <value>false</value>
+</property>
+<property>
+    <name>ozone.om.transport.class</name>
+    <value>org.apache.hadoop.ozone.om.protocolPB.Hadoop3OmTransportFactory</value>
+</property>
+
+<property>
+   <name>ipc.9862.callqueue.impl</name>
+   <value>org.apache.hadoop.ipc.FairCallQueue</value>
+</property>
+<property>
+   <name>ipc.9862.scheduler.impl</name>
+   <value>org.apache.hadoop.ipc.DecayRpcScheduler</value>
+</property>
+<property>
+   <name>ipc.9862.identity-provider.impl</name>
+   <value>org.apache.hadoop.ozone.om.OzoneIdentityProvider</value>
+</property>
+<property>
+   <name>ipc.9862.scheduler.priority.levels</name>
+   <value>2</value>
+</property>
+<property>
+   <name>ipc.9862.backoff.enable</name>
+   <value>true</value>
+</property>
+<property>
+   <name>ipc.9862.faircallqueue.multiplexer.weights</name>
+   <value>99,1</value>
+</property>
+<property>
+    <name>ipc.9862.decay-scheduler.thresholds</name>
+    <value>90</value>
+</property>
+```
+
+## Configuration for Ozone in Docker
+
+Under `hadoop-ozone/dist/target/ozone-*-SNAPSHOT/compose/ozone` 
+
+edit `docker-config` file and add the following
+
+```
+CORE-SITE.XML_ipc.9862.callqueue.impl=org.apache.hadoop.ipc.FairCallQueue
+CORE-SITE.XML_ipc.9862.scheduler.impl=org.apache.hadoop.ipc.DecayRpcScheduler
+CORE-SITE.XML_ipc.9862.identity-provider.impl=org.apache.hadoop.ozone.om.OzoneIdentityProvider
+CORE-SITE.XML_ipc.9862.scheduler.priority.levels=2
+CORE-SITE.XML_ipc.9862.backoff.enable=true
+CORE-SITE.XML_ipc.9862.faircallqueue.multiplexer.weights=99,1
+CORE-SITE.XML_ipc.9862.decay-scheduler.thresholds=90
+
+OZONE-SITE.XML_ozone.om.address=0.0.0.0:9862
```

Comment:
the `ozone.om.address` is already defined in the docker-config,
https://github.com/apache/ozone/blob/10811c597fb724060b6577d10a70bd638cfcd69c/hadoop-ozone/dist/src/main/compose/ozone/docker-config#L20
this effectively redefines it.  Also, in the ozone-secure environment it may be omitted altogether.  The configuration for the docker dev clusters should be looked at some more prior to including in the doc.  Suggest to open another jira to allow us to follow up?

Reply:
@neils-dev I removed the docker config altogether and filed a new jira for this: https://issues.apache.org/jira/browse/HDDS-7757